### PR TITLE
Another VIC20 TGI optimization

### PR DESCRIPTION
**Pull request by ops**
Seems like things moved forward during weekend...

I was able to save 3 more bytes using this:

https://github.com/cc65/cc65/pull/1078#issuecomment-657240602

### CODE REVIEW

**Comment by dmsc** [comment]
Hi!

> 
> I was able to save 3 more bytes using this:
> 
> [#1078 (comment)](https://github.com/cc65/cc65/pull/1078#issuecomment-657240602)

As I said before, you don't need to modify "tmp2", just avoid reloading Y in each loop.

And if SBASE is always page aligned, you can remove another two bytes by omiting the "LDA #<SBASE".

Have Fun!

**Comment by dmsc** [comment]
Hi Again,
 
> [#1078 (comment)](https://github.com/cc65/cc65/pull/1078#issuecomment-657240602)

Now, the code could be:

```
.proc INIT

; Initialize variables

        ldy     #$FF
        sty     BITMASK

; Make screen columns.

        lda     #<SBASE
        sta     tmp2
        .assert (<SBASE) = $0, error, "Error, SBASE must be page aligned"
        iny                     ; (ldy #$00)
        sty     tmp2+1
        sty     ERROR           ; Set to TGI_ERR_OK
        clc
        ldx     #$10

@NEXT_ROW:
        txa

@NEXT_COLUMN:
        sta     (tmp2),y
        iny
        adc     #ROWS
        bcc     @NEXT_COLUMN

; Step to next row on screen.
        inx
        cpx     #ROWS+$10
        bcc     @NEXT_ROW
```

**Comment by ops** [comment]
Thanks! And now low/high bytes fixed too!

File: libsrc/vic20/tgi/vic20-hi.s [hunk 1]
```diff
@@ -250,23 +251,21 @@ PATTERN_SOLID:
...
 
-        lda     #<SBASE
-        sta     tmp2
         lda     #>SBASE
         sta     tmp2+1
-        inx                     ; (ldx #$00)
-        stx     ERROR           ; Set to TGI_ERR_OK
+        iny                     ; (ldy #$00)
+        sty     tmp2
+        sty     ERROR           ; Set to TGI_ERR_OK
```
Comment (greg-king5) [inline libsrc/vic20/tgi/vic20-hi.s:263]: Style nitpick: I prefer that the
```asm
iny
sty ERROR
```
be placed above the comment "Make screen columns."  And, the `sty tmp2` should be put where the `sta tmp2` was.

**Comment by oliverschmidt** [comment]
@greg-king5: As you have reviewed this PR I ask you to merge as you see fit.

File: libsrc/vic20/tgi/vic20-hi.s (hunk 1 above)
Comment (ops) [reply libsrc/vic20/tgi/vic20-hi.s:263]: Fixed

Comment (greg-king5) [reply libsrc/vic20/tgi/vic20-hi.s:263]: It really is the `sty ERROR` that needs to be moved, it isn't a part of the "Make screen columns" section.

Comment (ops) [reply libsrc/vic20/tgi/vic20-hi.s]: Ah, right.

**Comment by greg-king5** [comment]
Thanks.

**Pull request merged**